core/src/main/java/org/infinispan/AdvancedCache.java

```diff
@@ -551,4 +551,17 @@
     * @return the entry set containing all of the CacheEntries
     */
    CacheSet<CacheEntry<K, V>> cacheEntrySet();
+
+   /**
+    * Attempts to remove the entry if it is expired.  Due to expired entries not being consistent across nodes, this
+    * will still attempt to remove the value if it is not present.  Note that this will raise an expired event
+    * even if the entry is not present.  Normally this method should never be invoked except by the
+    * {@link ExpirationManager}.
```

Comment:
I actually did manually invoke the command through the interceptor chain before, however this didn't support transactions.

I must admit I didn't look into using the FunctionalMap, but to be honest it wouldn't be very helpful timing wise for me as I have to backport this stuff as well, which doesn't have the FunctionalMap :(